extensions/ql-vscode/test/vscode-tests/cli-integration/variant-analysis/variant-analysis-manager.test.ts

```diff
@@ -202,5 +204,136 @@ describe("Variant Analysis Manager", () => {
 
       await expect(promise).rejects.toThrow(UserCancellationException);
     });
+
+    describe("check variant analysis generated packs", () => {
+      beforeEach(() => {
+        mockSubmitVariantAnalysis = jest
+          .spyOn(ghApiClient, "submitVariantAnalysis")
+          .mockResolvedValue({
+            id: 1,
+            query_language: QueryLanguage.Javascript,
+            query_pack_url: "http://example.com",
+            created_at: "2021-01-01T00:00:00Z",
+            updated_at: "2021-01-01T00:00:00Z",
+            status: "in_progress",
+            controller_repo: {
+              id: 1,
+              name: "vscode-codeql",
+              full_name: "github/vscode-codeql",
+              private: false,
+            },
+            actions_workflow_run_id: 20,
+            scanned_repositories: [] as any[],
+          });
+
+        executeCommandSpy = jest.spyOn(commands, "executeCommand");
+      });
+
+      it("should run a remote query that is part of a qlpack", async () => {
+        await doVariantAnalysisTest({
+          queryPath: "data-remote-qlpack/in-pack.ql",
+          filesThatExist: ["in-pack.ql", "lib.qll"],
+          filesThatDoNotExist: [],
+          qlxFilesThatExist: ["in-pack.qlx"],
+        });
+      });
+
+      it("should run a remote query that is not part of a qlpack", async () => {
+        await doVariantAnalysisTest({
+          queryPath: "data-remote-no-qlpack/in-pack.ql",
+          filesThatExist: ["in-pack.ql"],
+          filesThatDoNotExist: ["lib.qll", "not-in-pack.ql"],
+          qlxFilesThatExist: ["in-pack.qlx"],
+        });
+      });
+
+      it("should run a remote query that is nested inside a qlpack", async () => {
+        await doVariantAnalysisTest({
+          queryPath: "data-remote-qlpack-nested/subfolder/in-pack.ql",
+          filesThatExist: ["subfolder/in-pack.ql", "otherfolder/lib.qll"],
+          filesThatDoNotExist: ["subfolder/not-in-pack.ql"],
+          qlxFilesThatExist: ["subfolder/in-pack.qlx"],
+        });
+      });
+    });
+
+    async function doVariantAnalysisTest({
```

Comment:
An alternative for creating a function for this would be to use [`it.each`](https://jestjs.io/docs/api#each). That would avoid the need to create a function, which might make the tests easier to read (since all expectations are located within the `it`).

Reply:
This might work, but I'll have to see since in a follow-up PR I'm adding a new test that requires a bit of work before running the assertions. 

Actually...I think I can get this done by passing in a closure that can be invoked at the start of the `.each` function.

Reply:
Hmmmm...I don't think this is a great approach:

1. it prevents you from adding custom names for each test
2. it is more cumbersome to run individual tests

Reply:
It's totally fine to not do it, it was just something we might be able to try. Both look like valid points, so let's just keep it like this for now.